lib/jekyll/page.rb

```diff
@@ -123,12 +123,12 @@ def process(name)
     # layouts      - The Hash of {"name" => "layout"}.
     # site_payload - The site payload Hash.
     #
-    # Returns nothing.
+    # Returns String rendered page.
     def render(layouts, site_payload)
-      site_payload["page"] = to_liquid
-      site_payload["paginator"] = pager.to_liquid
-
-      do_layout(site_payload, layouts)
+      self.output = _renderer.tap do |renderer|
+        renderer.layouts = layouts
+        renderer.payload = site_payload
+      end.run
```

Comment:
@parkr since we're not removing Convertible at least till Jekyll 4, maybe I should just rollback this change and just call `do_layout` as it was before?
Because now it just duplicates part of the `Convertible.do_layout`